src/cpp/src/lora/adapter.cpp

```diff
@@ -342,17 +377,45 @@ void deduce_input_output_dims(NodePtr node, ov::Dimension& input_dim, ov::Dimens
 
 using LoRAVarMap = std::map<std::string, LoRAVarIDs>;
 
+struct BaseStateGetter {
+    std::shared_ptr<ov::Model> model;
+
+    BaseStateGetter(std::shared_ptr<ov::Model> model) : model(model) {}
+
+    NodePtr add_variable(const ov::op::util::VariableInfo& variable_info) const {
```

Comment:
It seems that the intention is to reuse `add_variable()` in two classes. That forced you to build an inheritance tree. Make `add_variable()` to accept `ov::Model` as the second arg and move the function to anonymous namespace. That should decouple the classes